server/src/Logger.hs

```diff
@@ -3,46 +3,63 @@ module Logger
   Logger,
   LogRecord,
   LogQueue,
+  LogLevel(..),
   newLogger,
   postLog,
   postLogBlocking,
   postStop,
-  processLogRecords
+  processLogRecords,
+  loggerSentryService
 )
 where
 
-import Control.Monad (unless)
+import SentryLogging
+import Config
+
+import Control.Monad (unless, when)
 import Control.Concurrent.STM (atomically)
 import Control.Concurrent.STM.TBQueue (TBQueue, newTBQueue, readTBQueue, writeTBQueue, isFullTBQueue)
-import Data.Text (Text)
-import GHC.Natural (Natural)
+import Data.Text (Text, unpack)
+import Data.Maybe (isJust)
 import Prelude hiding (log)
 
+
+
+import qualified System.Log.Raven.Types as Sentry
+
 import qualified Data.Text.IO as T
 
 type LogRecord = Text
+
+data LogLevel = LogInfo | LogError
+  deriving (Eq, Ord, Show, Read)
+
 type LogQueue = TBQueue LogCommand
 
-data Logger = Logger { loggerQueue :: LogQueue }
+data Logger = Logger { loggerQueue :: LogQueue, loggerSentryService :: Maybe Sentry.SentryService }
 
-data LogCommand = LogRecord LogRecord | LogStop
+data LogCommand = LogRecord LogLevel LogRecord | LogStop
   deriving (Eq, Ord, Show, Read)
 
-newLogger :: Natural -> IO Logger
-newLogger queueSize = Logger <$> atomically (newTBQueue queueSize)
+newLogger :: Config -> IO Logger
+newLogger config = Logger
+  <$> atomically (newTBQueue (fromIntegral $ configQueueCapacity config))
+  <*> if configDisableSentryLogging config then
+        pure Nothing
+      else maybe (pure Nothing) (fmap Just . getCrashLogger) (configSentryDSN config)
```

Comment:
This would be cleaner with a where clause, i.e.

```haskell
newLogger config = Logger
  <$> createQueue
  <*> createSentryService
  where
    createQueue = ...
    createSentryService = ...
```